### PR TITLE
Userstory 17 pages

### PR DESCRIPTION
**Pull request by nbinderHHN**
Complete feature of the page model.

### CODE REVIEW

File: src/de/hhn/aib/labsw/blackmirror/controller/Page.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+     */
+    protected void setWidgetsVisible() {
+        widgetsOnPage.forEach(w -> w.setVisible(true)); //Sets each widget visible
+    }
+
+    /**
+     * Updates all visible widgets on the page every second.
+     * Method coded from Markus Marewitz.
+     */
+    protected void updatePage() {
```
Comment (Code-Mozart) [inline src/de/hhn/aib/labsw/blackmirror/controller/Page.java:38]: This should therefore also be called onNextSecond

File: src/de/hhn/aib/labsw/blackmirror/controller/PageController.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
+package de.hhn.aib.labsw.blackmirror.controller;
+
+import de.hhn.aib.labsw.blackmirror.view.widgets.AbstractWidget;
+
+import java.util.ArrayList;
+
+/**
+ *  PageController Class to hold all pages and provide all necessary methods.
```
Comment (Code-Mozart) [inline src/de/hhn/aib/labsw/blackmirror/controller/PageController.java:8]: missing author and version

File: src/de/hhn/aib/labsw/blackmirror/controller/PageController.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+public class PageController
+{
+    private ArrayList<Page> pages;
+    private int pageIndex = 0;
+    private boolean isStandby = false;
+
+    public PageController ()
+    {
+        pages = new ArrayList<>();
+
```
Comment (Code-Mozart) [inline src/de/hhn/aib/labsw/blackmirror/controller/PageController.java:19]: cleanup newline

File: src/de/hhn/aib/labsw/blackmirror/controller/PageController.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    /**
+     * Adds a new page at the given index.
+     * @param widgetsOnPage Widgets to be shown on the new page.
+     * @param pageIndex Index where the page should be.
+     */
+    protected void addPageAtIndex(int pageIndex, ArrayList<AbstractWidget> widgetsOnPage)
+    {
+        pages.add(pageIndex, new Page(widgetsOnPage));
+    }
```
Comment (Code-Mozart) [inline src/de/hhn/aib/labsw/blackmirror/controller/PageController.java:39]: add a newline

File: src/de/hhn/aib/labsw/blackmirror/controller/PageController.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    {
+        pages.remove(pageIndex);
+    }
+
+    /**
+     * Moves a page to another position in the list.
+     * @param currentPosition Index, where the page currently is.
+     * @param newPosition Index, where the page should be moved to.
+     */
+    protected void movePage (int currentPosition, int newPosition)
```
Comment (Code-Mozart) [inline src/de/hhn/aib/labsw/blackmirror/controller/PageController.java:54]: Perhaps replace the index with the reference (like so 'protected void movePage(Page page, int newPosition') or rename 'currentPosition' to 'pageIndex'.
In my opinion usually prefer the reference over the index of this reference.

File: src/de/hhn/aib/labsw/blackmirror/controller/PageController.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        pages.add(newPosition, pageToMove);
+    }
+
+    /**
+     * Navigates to the next page.
+     * If it is on the last page, it navigates from the last page to the first page.
+     */
+    protected void goToNextPage()
+    {
+        if (isStandby == false)
```
Comment (Code-Mozart) [inline src/de/hhn/aib/labsw/blackmirror/controller/PageController.java:67]: Why so explicit and not 'if (!isStandby)'?

File: src/de/hhn/aib/labsw/blackmirror/controller/PageController.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+            if (pageIndex==(pages.size()-1))
+            {
+                pageIndex = 0;
+            }   else {
+                pageIndex = pageIndex+1;
+            }
+            pages.get(oldIndex).setWidgetsInvisible();
+            getCurrentPage().setWidgetsVisible();
+        }
+    }
```
Comment (Code-Mozart) [inline src/de/hhn/aib/labsw/blackmirror/controller/PageController.java:79]: add a newline

File: src/de/hhn/aib/labsw/blackmirror/controller/PageController.java [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    /**
+     * Navigates to the previous page.
+     * If it is on the first page, it navigates from the first page to the last page.
+     */
+    protected void goToPreviousPage()
+    {
+        if (isStandby == false)
```
Comment (Code-Mozart) [inline src/de/hhn/aib/labsw/blackmirror/controller/PageController.java:86]: prefer '!isStandby'

File: src/de/hhn/aib/labsw/blackmirror/controller/PageController.java [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+            if (pageIndex==0)
+            {
+                pageIndex = pages.size()-1;
+            }   else {
+                pageIndex = pageIndex-1;
+            }
+            pages.get(oldIndex).setWidgetsInvisible();
+            getCurrentPage().setWidgetsVisible();
+        }
+    }
```
Comment (Code-Mozart) [inline src/de/hhn/aib/labsw/blackmirror/controller/PageController.java:98]: consider merging both methods into one as they are very similar

**Review by Code-Mozart — CHANGES_REQUESTED** [review]
a view changes (sorry for beeing so nitpicky) and its good to go :)
most significantly: 'onNextSecond' shouldn't be renamed, it is more descriptive than 'update'

File: src/de/hhn/aib/labsw/blackmirror/controller/PageController.java (hunk 7 above)
Comment (nbinderHHN) [reply src/de/hhn/aib/labsw/blackmirror/controller/PageController.java:79]: > add a newline

I am a bit confused, what a new line should/would change

**Review by Code-Mozart — APPROVED** [review]

**Pull request merged**